rxn4chemistry/core.py

```diff
@@ -1295,3 +1307,147 @@ def update_reaction_settings(
+            }
+        }
+        response = requests.post(
+            self.routes.batch_executor_predict_from_request_url,
+            headers=self.headers,
+            data=json.dumps(data),
+            cookies={},
+        )
+        return response
+
+    @response_handling(success_status_code=200, on_success=default_on_success)
+    @ibm_rxn_api_limits
+    def predict_from_uri(
+        self, input_s3_path: str, output_s3_path: str,
+    ) -> None:
+        """
+        Run a batch prediction from a file in s3.
+
+        Args:
+            input_s3_path: S3 path for input file.
+            output_s3_path: S3 path for output file.
+
+        Examples:
+            Run a batch prediction using the wrapper:
+
+            >>> rxn4chemistry_wrapper.predict_from_uri(
+                {
+                "content": {
+                    "input_s3_path": "",
+                    "output_s3_path": ""
+                }
+            )
+        """
+        data = {
+            "content": {
+                "input_s3_path": input_s3_path,
+                "output_s3_path": output_s3_path
+            }
+        }
+        requests.post(
+            self.routes.batch_executor_predict_from_uri_url,
+            headers=self.headers,
+            data=json.dumps(data),
+            cookies={},
+        )
+
+    @response_handling(success_status_code=200, on_success=default_on_success)
+    @ibm_rxn_api_limits
+    def download_from_uri(
+        self, s3_path: str,
+    ) -> requests.models.Response:
+        """
+        Creates a presigned URI for downloading a file in S3.
+
+        Args:
+            s3_path: a s3 uri.
+
+        Returns:
+            A presigned URI for download.
+
+        Examples:
+            Run an example using the wrapper:
+
+            >>> rxn4chemistry_wrapper.download_from_uri({"s3_path": ""})
+        """
+        data = {
+            "s3_path": s3_path
+        }
+        response = requests.get(
+            self.routes.batch_executor_download_from_uri_url,
+            headers=self.headers,
+            data=json.dumps(data),
+            cookies={},
+        )
+        return response
+
+    @response_handling(success_status_code=200, on_success=default_on_success)
+    @ibm_rxn_api_limits
+    def read_from_uri(
+        self, s3_path: str,
+    ) -> requests.models.Response:
+        """
+        Read the content of an input/output file in S3.
+
+        Args:
+            s3_path: a s3 uri.
+
+        Returns:
+            The list of lines in the file.
+
+        Examples:
+            Run an example using the wrapper:
+
+            >>> rxn4chemistry_wrapper.read_from_uri({"s3_path": ""})
+        """
+        data = {
+            "s3_path": s3_path
+        }
+        response = requests.get(
+            self.routes.batch_executor_read_from_uri_url,
+            headers=self.headers,
+            data=json.dumps(data),
+            cookies={},
+        )
+        return response
```

Comment:
In case the batch executor url is not set, let's raise a ValueError notifying the users the endpoints are not configured correctly.